pkg/ring/model.go

```diff
@@ -79,14 +79,14 @@ func (d *Desc) RemoveIngester(id string) {
 // This method assumes that Ring is in the correct state, 'from' ingester has no tokens anywhere,
 // and 'to' ingester uses either normalised or non-normalised tokens, but not both. Tokens list must
 // be sorted properly. If all of this is true, everything will be fine.
-func (d *Desc) ClaimTokens(from, to string, normaliseTokens bool) []uint32 {
-	var result []uint32
+func (d *Desc) ClaimTokens(from, to string, normaliseTokens bool) Tokens {
+	result := Tokens{}
 
 	if normaliseTokens {
 
 		// If the ingester we are claiming from is normalising, get its tokens then erase them from the ring.
 		if fromDesc, found := d.Ingesters[from]; found {
-			result = fromDesc.Tokens
+			result = append(result, fromDesc.Tokens...)
```

Comment:
Why are we not doing `result = Tokens(fromDesc.Tokens)`? That doesn't allocate, but I think this does.

Reply:
Ah, this is something left behind with so many design changes in between. Will clean this up.